### ISSUE TITLE
Important ICS fails

### ISSUE THREAD

**Issue by mbrandis** [owncloud/calendar#856]
### Steps to reproduce

Command line:
1. Use georgegehrke script automatedimport.php to import an ICS file.
2. php automatedimport.php /var/www/owncloud/data/XXX/files/YYY.ics ZZZ 20 Europe/Berlin
3. Result: PHP Fatal error:  Call to a member function isEmpty() on null in /var/www/owncloud/apps/calendar/lib/app.php on line 146

Web Interface:
1. Click on ICS file.
2. Choose a calendar to import into and click to remove old entries.
3. Dialog becomes gray and nothing else happens.

### Expected behaviour
Import of ICS file should work.

### Actual behaviour
See above.

### Server configuration
**Operating system**: Debian Stable

**Web server:** Apache 2.4.10

**Database:** MySQL 5.5.43

**PHP version:** 5.6.9

**ownCloud version:** 8.0.5

**Updated from an older ownCloud or fresh install:** Update from 6 to 7 to 8

**List of activated apps:**

 - activity
 - bookmarks
 - calendar
 - contacts
 - documents
 - files
 - files_locking
 - files_mv
 - files_pdfviewer
 - files_sharing
 - files_texteditor
 - files_trashbin
 - files_versions
 - firstrunwizard
 - galleryplus
 - mozilla_sync
 - ownnote
 - sharelinks
 - user_sql

**The content of config/config.php:**

  'trusted_domains' => 
  array (
    0 => 'XXX',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'dbtype' => 'mysql',
  'version' => '8.0.5.2',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'owncloud',
  'installed' => true,
  'forcessl' => true,
  'loglevel' => '3',
  'log_authfailip' => true,
  'log_rotate_size' => 1048576,
  'default_language' => 'de',
  'theme' => '',
  'maintenance' => false,
  'trashbin_retention_obligation' => 7,
  'trashbin_auto_expire' => true,
  'share_folder' => '/Shared',
  'appstoreenabled' => true,
  'appstoreurl' => 'https://api.owncloud.com/v1',

**Are you using external storage, if yes which one:** local/smb/sftp/...

No

**Are you using encryption:** yes/no

No

**Are you using an external user-backend, if yes which one:** 

SQL via app user_sql

### Client configuration
**Browser:** Firefox 39

**Operating system:** Debian Testing

#### Web server error log

No entry when executing the task.

#### ownCloud log (data/owncloud.log)

No entry when executing the task.

#### Browser log

N/A

**Comment by shoeper** [comment]
can confirm this.

**Comment by mbrandis** [comment]
The import works partially. There are entries in the calendar after the PHP error.

**Comment by novalore** [comment]
Testing owncloud calendar, I noticed this behaviour too.
It's a very important feature for people migrating from other caldav implementations. please fix it
:+1:

**Comment by georgehrke** [comment]
please enable debug mode (see documentation) and provide the owncloud.log log file and the server's error log

**Comment by novalore** [comment]
I enabled debug logging, but it does not seem to be helpful at all:
Web server error log: 
No entry when executing the task. (Nginx + php-fpm)

ownCloud log (data/owncloud.log):
Debug	files_locking	INFO: Incrementing lock type 1 count for /var/www/owncloud/data/faust/files/tommaso19.ics	2015-08-21T07:20:47+00:00
Debug	files_locking	INFO: Incrementing lock type 1 count for /var/www/owncloud/data/faust/files/tommaso19.ics	2015-08-21T07:20:47+00:00
Debug	files_locking	PASS: Acquired read lock for /var/www/owncloud/data/faust/files/tommaso19.ics	2015-08-21T07:20:47+00:00

The progress bar fills immediately and nothing else happens (see screenshot)
![owncloud_sshot](https://cloud.githubusercontent.com/assets/1128726/9403348/85c39e9e-47e6-11e5-9cc1-d214e625179c.png)

**Comment by novalore** [comment]
My bad! 
I was trying to import an ics to a calendar my user has no write access to.
It looks like a UX bug, not a functional one: calendars the current user can't write on shoul not be listed in the import modal window, don't you think?

**Comment by shoeper** [comment]
> My bad! 
> I was trying to import an ics to a calendar my user has no write access to.
> It looks like a UX bug, not a functional one: calendars the current user can't write on should not be listed in the import modal window, don't you think?

what is write support good for when one wants to IMPORT an ics file? The plugin should just detect it and write to some tmp file if writing is necessary.

**Comment by novalore** [comment]
@shoeper data are imported from an ICS file to be written on a calendar. If you don't have write access to said calendar, you should not be able to import data targeting it.

**Comment by shoeper** [comment]
correct. Sorry I understood it the other way around and was quite confused.
